daml-foundations/daml-ghc/BUILD.bazel

```diff
@@ -14,74 +14,269 @@ load(
+        ":daml-ghc-compiler",
         "//compiler/daml-lf-ast",
         "//compiler/daml-lf-proto",
         "//compiler/daml-lf-tools",
         "//compiler/haskell-ide-core",
         "//compiler/scenario-service/client",
+        "//libs-haskell/da-hs-base",
+    ],
+)
+
+da_haskell_library(
+    name = "daml-service-daml-compiler",
+    srcs = glob([
+        "src/DA/Service/Daml/Compiler/**/*.hs",
+    ]),
+    hazel_deps = [
+        "base",
+        "bytestring",
+        "containers",
+        "directory",
+        "extra",
+        "filepath",
+        "ghc-lib-parser",
+        "http-types",
+        "mtl",
+        "network-uri",
+        "safe",
+        "tagged",
+        "text",
+        "time",
+        "transformers",
+        "zip-archive",
+    ],
+    src_strip_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":daml-development",
+        ":daml-ghc-compiler",
+        "//compiler/daml-lf-ast",
+        "//compiler/daml-lf-proto",
+        "//compiler/haskell-ide-core",
+        "//compiler/scenario-service/client",
+        "//libs-haskell/da-hs-base",
+    ],
+)
+
+da_haskell_library(
+    name = "daml-development-testing",
+    srcs = glob([
+        "src/Development/IDE/State/API/Testing.hs",
+    ]),
+    hazel_deps = [
+        "aeson",
+        "base",
+        "containers",
+        "directory",
+        "extra",
+        "filepath",
+        "haskell-lsp-types",
+        "mtl",
+        "stm",
+        "temporary",
+        "text",
+        "time",
+        "vector",
+    ],
+    src_strip_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":daml-development",
+        ":daml-ghc-compiler",
+        ":daml-service-daml-compiler",
+        "//compiler/haskell-ide-core",
+    ],
+)
+
+da_haskell_library(
+    name = "daml-service-daml-language-server",
+    srcs = glob([
+        "src/DA/Service/Daml/LanguageServer.hs",
+        "src/DA/Service/Daml/LanguageServer/**/*.hs",
+    ]),
+    hazel_deps = [
+        "aeson",
+        "async",
+        "base",
+        "containers",
+        "extra",
+        "managed",
+        "network-uri",
+        "safe",
+        "safe-exceptions",
+        "stm",
+        "tagged",
+        "uri-encode",
+    ],
+    src_strip_prefix = "src",
+    visibility = ["//visibility:public"],
+    deps = [
+        ":daml-service-daml-compiler",
+        "//compiler/daml-lf-ast",
+        "//compiler/haskell-ide-core",
         "//daml-assistant:daml-project-config",
-        "//daml-lf/archive:daml_lf_haskell_proto",
-        "//libs-haskell/bazel-runfiles",
         "//libs-haskell/da-hs-base",
         "//libs-haskell/da-hs-language-server",
-        "//libs-haskell/prettyprinter-syntax",
+    ],
+)
+
+# This target takes a while to compile, probably due to TemplateHaskell.
+# It's separated into its own target not to inflict that compilation
+# time on changes outside the DA.Daml.GHC.Damldoc.Types module
+da_haskell_library(
```

Comment:
Can we put all of damldoc into this library?

Reply:
The rest of `damldoc` is in [the target called `daml-ghc-damldoc`](https://github.com/digital-asset/daml/pull/1165/files#diff-48f80e007f92d8454827e7e5ea770aa8R214). I'm guessing you'd want all of `damldoc` in that target.

Sure, we can do that. However, as the comment says
```
# This target takes a while to compile, probably due to TemplateHaskell.
# It's separated into its own target not to inflict that compilation
# time on changes outside the DA.Daml.GHC.Damldoc.Types module
```

But, if you prefer to not separate out `Damldoc.Types` knowing that cost, I'm happy to keep it in there.

Reply:
Putting a single module into a package seems quite extreme to me and is something I'd like to avoid. Let's move it to the rest of of `damldoc`. We can probably improve compilation time by removing the Template Haskell.